### PR TITLE
Provide mechanism to configure XContent parsing constraints (after update to Jackson 2.15.0 and above)

### PR DESCRIPTION
**Pull request by reta**
### Description
Provide mechanism to configure XContent parsing constraints (after update to Jackson 2.15.0 and above)

### Related Issues
Resolves #[7549](https://github.com/opensearch-project/OpenSearch/issues/7549)

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@andrross @dblock curious what do you think folks, @martin-gaievski @ylwu-amzn fyi

File: libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java
Comment (andrross) [inline libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java:95]: Where does the "3145730" value come from?

File: libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java
Comment (andrross) [inline libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java:161]: Why not fall through here?

File: libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java
Comment (reta) [reply libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java:95]: That's from YAML parser error :)

File: libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java
Comment (reta) [reply libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java:161]: Copy / paste from other test, will fix that, thanks!

File: libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java
Comment (reta) [reply libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java:95]: https://github.com/snakeyaml/snakeyaml/blob/master/src/main/java/org/yaml/snakeyaml/LoaderOptions.java#L39

**Comment by martin-gaievski** [comment]
@andrross could you please review again, seems that Andriy (reta) has been addressed your previous comments? This change should unblock several plugins that are blocked by jackson 2.15 upgrade

**Review by andrross — APPROVED** [review]

**Comment by andrross** [comment]
@reta Sorry I missed this prior to merging, but since this introduces a new setting for cluster operators I think this merits a changelog entry. What do you think?

**Comment by reta** [comment]
> @reta Sorry I missed this prior to merging, but since this introduces a new setting for cluster operators I think this merits a changelog entry. What do you think?

I will do that (in a moment) and also include the documentation issue, thank you

**Pull request merged**
